// Copyright (C) 2017 go-nebulas authors
//
// This file is part of the go-nebulas library.
//
// the go-nebulas library is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// the go-nebulas library is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with the go-nebulas library.  If not, see <http://www.gnu.org/licenses/>.
//

syntax = "proto3";

package nvm;

service NVMService{

    rpc DeploySmartContract(NVMDeployRequest) returns (NVMResponse) {}

    rpc NVMDataExchange(stream NVMDataRequest) returns (stream NVMResponse) {}

}

message NVMDeployRequest{
    string script_src = 1;
    string from_addr = 2;
    int64 block_height = 3;
    string type = 4;
}

message NVMDataRequest{
    string request_type = 1;
    string script_src = 2;
    string function_name = 3;
    string contract_addr = 4;
    string extra_data = 5;
}

message NVMResponse {
    int32 result = 1;
    string msg = 2;
}
